code/controllers/subsystems/initialization/materials.dm

```diff
@@ -146,6 +146,21 @@ SUBSYSTEM_DEF(materials)
 		global.default_strata_type_by_z[s_key] = pick(subtypesof(/decl/strata))
 	return global.default_strata_type_by_z[s_key]
 
+/datum/controller/subsystem/materials/proc/get_material_by_name(var/mat_name)
+	RETURN_TYPE(/decl/material)
+	if(!mat_name)
+		return
+	mat_name = lowertext(mat_name)
+	if(isnull(materials_by_name[mat_name]))
+		var/list/mats = decls_repository.get_decls_of_subtype(/decl/material)
+		for(var/mat in mats)
+			var/decl/material/mat_decl = mats[mat]
+			if(lowertext(mat_decl.name) == mat_name)
+				materials_by_name[mat_name] = mat_decl
+				return mat_decl
+		materials_by_name[mat_name] = FALSE
```

Comment:
Why not just create this list once (either at init or the first time this proc is called), putting in all the decls? This is slower and also has a questionable return type.

Reply:
This was a partial commit I meant to snip out later when I realized materials_by_name existed but was incorrectly populated. Gone now.